From be2bbc4da04cfa08b5754307b2f386e4ec062a1a Mon Sep 17 00:00:00 2001
From: Pramila Bishnoi <bishnoi.pramila.rg@gmail.com>
Date: Fri, 5 Jun 2026 20:01:19 +0530
Subject: [PATCH] fix(common): fix invalid unary expression in uint128_fallback
 subtraction

Signed-off-by: Pramila Bishnoi <bishnoi.pramila.rg@gmail.com>
---
 include/common/int128.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/common/int128.h b/include/common/int128.h
index 40f49e7270e8..168f488050dd 100644
--- a/include/common/int128.h
+++ b/include/common/int128.h
@@ -581,8 +581,8 @@ operator+(detail::uint128_fallback LHS, unsigned int RHS) {
 
 inline constexpr detail::uint128_fallback
 operator-(unsigned int LHS, detail::uint128_fallback RHS) {
-  uint128_fallback Result = RHS;
-  return (~Result) + 1 + LHS;
+  detail::uint128_fallback Negated = {~RHS.high(), ~RHS.low()};
+  return Negated + 1 + LHS;
 }
 
 inline constexpr detail::uint128_fallback &
